src/Avalonia.Input/PenDevice.cs

```diff
@@ -0,0 +1,415 @@
+using System;
+using System.Linq;
+using System.Reactive.Linq;
+using Avalonia.Input.Raw;
+using Avalonia.Interactivity;
+using Avalonia.Platform;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Input
+{
+    /// <summary>
+    /// Represents a pen/stylus device.
+    /// </summary>
+    public class PenDevice : IPenDevice, IDisposable
+    {
+        private int _clickCount;
+        private Rect _lastClickRect;
+        private ulong _lastClickTime;
+
+        private readonly Pointer _pointer;
```

Comment:
What happens if user has multiple tablets attached to their device?

Reply:
This is a super corner case.

Reply:
Microsoft Surface Pro with a separate Wacom tablet connected to a dock station.